### PR TITLE
Recognize ID type by field descriptor

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Extends `Identifier.Type` with an ability to match itself to a `FieldDescriptor`.
  * Removes `MessageField` class, which is no longer used by `core-java`.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, @dmdashenkov, PTAL.

File: base/src/main/java/io/spine/base/Identifier.java [hunk 1]
```diff
@@ -317,6 +322,42 @@ private Any pack() {
...
+                               return false;
+                           }
+                           if (idType == Type.MESSAGE) {
+                               @SuppressWarnings("unchecked") // safe since it's Message type.
+                               TypeUrl messageType = TypeUrl.of(
+                                       (Class<? extends Message>) idClass);
+                               TypeUrl fieldType = TypeUrl.from(f.getMessageType());
+                               return fieldType.equals(messageType);
+                           }
+                           return true;
```
Comment (dmdashenkov) [inline base/src/main/java/io/spine/base/Identifier.java]: Please split into several separate `filter` calls for better readability.

File: base/src/main/java/io/spine/base/Identifier.java [hunk 2]
```diff
@@ -500,6 +567,8 @@ Message fromMessage(Message message) {
 
         abstract <I> I defaultValue(Class<I> idClass);
 
+        abstract boolean matchField(FieldDescriptor field);
```
Comment (dmdashenkov) [inline base/src/main/java/io/spine/base/Identifier.java]: Let's use `FieldDeclaration` instead of a descriptor.

Comment (alexander-yevsyukov) [reply base/src/main/java/io/spine/base/Identifier.java]: As discussed vocally let's keep it as is to simplify filtering of fields.

**Pull request merged**